inference-engine/src/transformations/src/transformations/common_optimizations/algebraic_simplification.cpp

```diff
@@ -37,12 +37,11 @@ static bool simplify_gather(std::shared_ptr<Node> node) {
             return false;
         }
 
-
-        if (!gather->is_axis_set()) {
+        auto axis = gather->get_axis();
+        if (axis == opset3::Gather::AXIS_NOT_SET_VALUE) {
```

Comment:
Am I right that you have changed the public API of Gather from opset3 and removed public method `is_axis_set()`?

Reply:
No, previously when I was enabling Gather-7 I added `is_axis_set()` https://github.com/openvinotoolkit/openvino/pull/4885/files#diff-1899273196bcbed10cce05f78cfe5281ab7b190af61e39452f4ab6eb7c774c28R82 
and now reverted back to `AXIS_NOT_SET_VALUE` to match old public API.

Here are status far before my changes for Gather-7, `AXIS_NOT_SET_VALUE` was present there
https://github.com/openvinotoolkit/openvino/blob/35bff0d8a1e4c7fb6ab55ff8948a9bc1019b218c/inference-engine/src/transformations/src/transformations/common_optimizations/algebraic_simplification.cpp#L52-L57

https://github.com/openvinotoolkit/openvino/blob/35bff0d8a1e4c7fb6ab55ff8948a9bc1019b218c/ngraph/core/src/op/gather.cpp#L34

https://github.com/openvinotoolkit/openvino/blob/35bff0d8a1e4c7fb6ab55ff8948a9bc1019b218c/ngraph/core/include/ngraph/op/gather.hpp#L31

I personally think that `is_axis_set()` is better but if we cannot change public API then I do not insist on solution with `is_axis_set()`